Microsoft.Toolkit.Uwp.UI.Controls.Input/TokenizingTextBox/TokenizingTextBox.cs

```diff
@@ -77,6 +77,16 @@ private void ItemsSource_PropertyChanged(DependencyObject sender, DependencyProp
             if (ItemsSource != null && ItemsSource.GetType() != typeof(InterspersedObservableCollection))
             {
                 _innerItemsSource = new InterspersedObservableCollection(ItemsSource);
+
+                if (ReadLocalValue(MaxTokensProperty) != DependencyProperty.UnsetValue && _innerItemsSource.ItemsSource.Count > MaxTokens)
+                {
+                    // Reduce down to the max as necessary.
+                    for (var i = _innerItemsSource.ItemsSource.Count - 1; i >= Math.Max(MaxTokens, 0); --i)
```

Comment:
Is the Math.Max in case of negative `int `values for `MaxTokens`?

Reply:
Yup!

Reply:
`uint `is an option - similar code exists here, for the Rank property of `DataGridDataItem `in the `DataGrid`: 
![image](https://user-images.githubusercontent.com/86266896/128782734-fe8e2609-b299-47f1-9024-adde4c2613ce.png)

Reply:
`DataGridDataItem` isn't a DependencyObject, but we can try it. I see it being used in the Custom Cursor property for GridSplitter. I can also try it in the `ConstrainedBox`.

Reply:
Nope XAML doesn't like `uint` :(

![image](https://user-images.githubusercontent.com/24302614/128783718-659e7239-0f02-4688-965e-10452f1e99e6.png)

The GridSplitter one probably needs to be updated too. Funnily enough a [WinUI issue was just filed](https://github.com/microsoft/microsoft-ui-xaml/issues/5639).

Reply:
It's not necessarily as intuitive up front, using `int`, since I have to manage potential negatives. But it does support the usage of `-1` to mean none/empty, which is kind of a nice thing.